src/modules/publishProduct/reducer.ts

```diff
@@ -0,0 +1,22 @@
+import * as D from '../../definitions'
+import * as Redux from 'redux'
+
+const initialState: D.PublishProductsState = {
+  currentImageUrl: '',
+}
+
+const publishProductReducer: Redux.Reducer<D.PublishProductsState> =
+  (state: D.PublishProductsState, action: D.GeneralAction): D.PublishProductsState => {
+    state = state || initialState
+    switch (action.type) {
+      case 'UPLOAD_IMAGE_SUC':
```

Comment:
It's better to make action & reducer use the same type. 
Otherwise `export const UPLOAD_IMAGE_SUC = 'UPLOAD_IMAGE_SUC'` looks useless.